src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -35,6 +35,8 @@ public class PackageBuilder : IPackageMetadata
         /// </summary>
         public const int MaxIconFileSize = 1024 * 1024;
 
+        public IDictionary<string, string> AliasMappings { get; set; }
```

Comment:
I don't know if we're under too much time pressure to change this now, but I think `PackageBuilder` already has enough APIs for the feature to be implememented without adding new APIs to `PackageBuilder`.

The `Files` property is a collection of `IPackageFile`, and `IPackageFile` and the `Path` property's XML doc says "Gets the full path of the file inside the package". If `NuGet.Commands` set this to the transformed/mapped path, I'm not sure you'd need to make any modifications to `PackageBuilder`.

Reply:
We still need it for the warning, to make sure something is actually an alias.